### PR TITLE
Template not updating due to git diff misuse

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I fixed the CI action for updating templates from the source.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: .github/workflows/update-e2e-template.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
       - name: Check for changes
         id: check_changes
         run: |
-          if git diff --quiet "${{ github.event.pull_request.head.ref }}" "origin/${{ github.event.pull_request.head.ref }}"; then
+          if git diff --quiet "origin/${{ github.event.pull_request.head.ref }}"; then
```
Comment (fa9r) [inline .github/workflows/update-e2e-template.yml:73]: so this is the main change?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,136 +16,136 @@ concurrency:
   cancel-in-progress: true
```
Comment (fa9r) [inline .github/workflows/ci.yml:16]: this file should probably be reverted before merging

File: .github/workflows/update-e2e-template.yml (hunk 1 above)
Comment (avishniakov) [reply .github/workflows/update-e2e-template.yml:73]: Yes and should be the only one. I opened this one but forgot to mention that it is not review-ready, I just needed CIs to run for validation.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (avishniakov) [reply .github/workflows/ci.yml:16]: Absolutely, this is WIP, just need CIs, thus a normal PR

**Comment by avishniakov** [comment]
Template update logic validated isolatedly and CI brought in [c80e34b](https://github.com/zenml-io/zenml/pull/1844/commits/c80e34bd95895455c59d52e3508d555c2c8c065b)

File: .github/workflows/update-e2e-template.yml (hunk 1 above)
Comment (avishniakov) [reply .github/workflows/update-e2e-template.yml:73]: It is ready 👍🏼

**Review by fa9r — APPROVED** [review]

**Pull request merged**